### PR TITLE
consider entries on inactive journals for the general ledger report

### PR DESCRIPTION
**Pull request by mdietrichc2c**
On a customer's instance, the entries that used an inactive journal (before
it was deactivated) were not considered in the general ledger; this fixes
the behaviour.

### CODE REVIEW

**Comment by StefanRijnhart** [comment]
Thanks! Looks very useful. Maybe I am wrong, but I see no obvious way to check the status of the upstream pull request of this branch into Odoo (I cannot even click on the proposed branch to navigate to its github page). Maybe you can post a link to the upstream pull request in this pull request's description?

File: addons/account/wizard/account_report_common.py [hunk 1]
```diff
@@ -136,7 +136,12 @@ def _get_fiscalyear(self, cr, uid, context=None):
         return fiscalyears and fiscalyears[0] or False
 
     def _get_all_journal(self, cr, uid, context=None):
-        return self.pool.get('account.journal').search(cr, uid ,[])
+        # The 'active_test' context is used, because reports must
+        # return entries created on inactive journals.
+        return self.pool.get('account.journal').search(cr, uid, [],
+                                                       context={
+                                                         'active_test': False
+                                                       })
```
Comment (StefanRijnhart) [inline addons/account/wizard/account_report_common.py]: Please keep any existing context by using something like context=dict(context or {}, active_test=False)

File: addons/account/wizard/account_report_common.py [hunk 2]
```diff
@@ -174,7 +179,17 @@ def check_report(self, cr, uid, ids, context=None):
         data = {}
         data['ids'] = context.get('active_ids', [])
         data['model'] = context.get('active_model', 'ir.ui.menu')
-        data['form'] = self.read(cr, uid, ids, ['date_from',  'date_to',  'fiscalyear_id', 'journal_ids', 'period_from', 'period_to',  'filter',  'chart_account_id', 'target_move'], context=context)[0]
+        data['form'] = self.read(cr, uid, ids, ['date_from', 'date_to',
+                                                'fiscalyear_id', 'period_from',
+                                                'period_to', 'filter',
+                                                'chart_account_id',
+                                                'target_move'])[0]
+        # The 'active_test' context is used, because reports must
```
Comment (StefanRijnhart) [inline addons/account/wizard/account_report_common.py:187]: Looks like nothing changed in this line. Can you revert the changes in whitespace, to prevent unnecessary conflicts with upstream?

File: addons/account/wizard/account_report_common.py [hunk 3]
```diff
@@ -174,7 +179,17 @@ def check_report(self, cr, uid, ids, context=None):
...
+        # return entries created on inactive journals.
+        data['form']['journal_ids'] = self.read(cr, uid, ids, ['journal_ids'],
+                                                context={
```
Comment (StefanRijnhart) [inline addons/account/wizard/account_report_common.py]: I know that 'active_test' works in search operations. Is it meaningful in read operations as well?

**Comment by clonedagain** [comment]
@StefanRijnhart Isn't the "active" state on journals one of the fields added by OCB, which caused Fabien to refuse to merge OCB into core in the first place? I think exactly because he feared this sort of side-effects.
So I guess there can not be an upstream PR.

**Comment by mdietrichc2c** [comment]
@StefanRijnhart as @clonedagain said, the field "active" was added on journal in OCB only, not Odoo (https://code.launchpad.net/~camptocamp/ocb-addons/7.0-fix-bug-1180803).
That's the reason I didn't make an upstream PR; only OCB is impacted.

File: addons/account/wizard/account_report_common.py (hunk 2 above)
Comment (mdietrichc2c) [inline addons/account/wizard/account_report_common.py:187]: It was for (future) PEP8 validation; I can revert it if needed.

File: addons/account/wizard/account_report_common.py (hunk 3 above)
Comment (mdietrichc2c) [inline addons/account/wizard/account_report_common.py]: It actually is; without that, if an inactive ID is present in the object's one2many, the value returned will not include the inactive one.

File: addons/account/wizard/account_report_common.py (hunk 1 above)
Comment (mdietrichc2c) [inline addons/account/wizard/account_report_common.py]: Thanks, I'll fix this ASAP.

File: addons/account/wizard/account_report_common.py (hunk 2 above)
Comment (StefanRijnhart) [inline addons/account/wizard/account_report_common.py:187]: Thanks. I now see that you read the journal_ids separately with the active_test context key, but I still wonder if that makes a difference.

**Comment by StefanRijnhart** [comment]
Thanks for the info about having no upstream PR.

File: addons/account/wizard/account_report_common.py (hunk 3 above)
Comment (StefanRijnhart) [inline addons/account/wizard/account_report_common.py]: Oh dear, I remember coming across that indeed. Thanks for the info.

**Comment by StefanRijnhart** [comment]
:+1:

File: addons/account/wizard/account_report_common.py [hunk 4]
```diff
@@ -136,7 +136,11 @@ def _get_fiscalyear(self, cr, uid, context=None):
         return fiscalyears and fiscalyears[0] or False
 
     def _get_all_journal(self, cr, uid, context=None):
-        return self.pool.get('account.journal').search(cr, uid ,[])
+        # The 'active_test' context is used, because reports must
+        # return entries created on inactive journals.
+        return self.pool.get('account.journal').search(
```
Comment (pedrobaeza) [inline addons/account/wizard/account_report_common.py]: You can put `self.pool['account.journal']` for shorter line

File: addons/account/wizard/account_report_common.py [hunk 5]
```diff
@@ -174,7 +178,12 @@ def check_report(self, cr, uid, ids, context=None):
         data = {}
         data['ids'] = context.get('active_ids', [])
         data['model'] = context.get('active_model', 'ir.ui.menu')
-        data['form'] = self.read(cr, uid, ids, ['date_from',  'date_to',  'fiscalyear_id', 'journal_ids', 'period_from', 'period_to',  'filter',  'chart_account_id', 'target_move'], context=context)[0]
+        data['form'] = self.read(cr, uid, ids, ['date_from', 'date_to', 'fiscalyear_id', 'period_from', 'period_to', 'filter', 'chart_account_id', 'target_move'], context=context)[0]
```
Comment (pedrobaeza) [inline addons/account/wizard/account_report_common.py]: As this is not going to be included upstream, please make your new lines PEP8 compliant (80 cols limit).

**Comment by mdietrichc2c** [comment]
@pedrobaeza okay, I re-added PEP8 validation and modified the self.pool (is it always better to use this instead of .get() now?)

**Comment by clonedagain** [comment]
> it always better to use this instead of .get() now?
> Yes it raises meaningful errors if something goes wrong, so it makes 
> debugging easier and costs nothing.

**Comment by gurneyalex** [comment]
:+1:
